### PR TITLE
Sending referenced works to datacite

### PR DESCRIPTION
**Pull request by Harry-Jisc**
The purpose of this PR is to send the references to datacite so they are retained on the DOI's data. This PR contains the following changes: 
- The references are passed into the `updateDOI` helper function in the api 
- The references with DOIs are combined with the linked publications and are sent as `relatedIdentifiers`. Any references without DOIs are sent as `relatedItems` 
- The `updateDOINames` script's name has been changed to `updateDoi` and the references are being pulled and passed through the `updateDOI` script

---

### Acceptance Criteria:

As Per [OCT-611](https://jiscdev.atlassian.net/browse/OCT-611?atlOrigin=eyJpIjoiZjIwODQ3MjhiN2E5NDUxYzk3ZTI2NmJmZWE5OWY1YzUiLCJwIjoiaiJ9)

---

### How to test this PR:
- Create a publication with several references (both with and without DOIs) 
- Publish 
- Login to [our test repository](https://doi.test.datacite.org/sign-in) and search for the relevant DOI
- Check that this data has been retained in the JSON object under  `relatedIdentifiers` and `relatedItems` 

---

### Screenshots

https://user-images.githubusercontent.com/100135505/235718524-456f545a-3005-441a-a754-51fdc1672fe3.mov

[OCT-611]: https://jiscdev.atlassian.net/browse/OCT-611?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: api/src/lib/helpers.ts [hunk 1]
```diff
@@ -77,6 +86,58 @@ export const updateDOI = async (doi: string, publication: I.PublicationWithMetad
...
+                  relatedIdentifier: relatedIdentifier.publicationToRef.doi,
+                  relatedIdentifierType: 'DOI',
+                  relationType: 'Continues'
+              }
+    );
+
+    const doiReferences = references.map((reference) => {
+        if (reference.type !== 'DOI' || !reference.location) return;
+
+        const doi: any = getFullDOIsStrings(reference.location);
```
Comment (Harry-Jisc) [inline api/src/lib/helpers.ts:106]: eslint is warning about the use of any type here, better to use something else or allow it to infer from the return type of getFullDOIsStrings

File: api/src/lib/interface.ts [hunk 2]
```diff
@@ -571,6 +571,12 @@ export interface DataCiteCreator {
     nameIdentifiers: DataCiteCreatorNameIdentifiers[];
 }
 
+export interface DataCiteDoiReferences {
```
Comment (Harry-Jisc) [inline api/src/lib/interface.ts:574]: This can be removed, it's not being used anywhere.

Comment (shaunajisc) [reply api/src/lib/interface.ts]: Good spot - I've deleted it ☺️

File: api/src/lib/helpers.ts [hunk 3]
```diff
@@ -77,6 +86,58 @@ export const updateDOI = async (doi: string, publication: I.PublicationWithMetad
         if (coAuthor.user !== null) return createCreatorObject(coAuthor.user);
     });
 
+    const linkedPublications = publication?.linkedTo.map((relatedIdentifier) =>
```
[the remark this answers is not in the document]
Comment (shaunajisc) [reply api/src/lib/helpers.ts]: Much cleaner thanks - I've amended that

**Review by finlay-jisc — APPROVED** [review]

**Pull request merged**